builtin/mainmenu/tab_multiplayer.lua

```diff
@@ -92,17 +115,23 @@ end
 
 --------------------------------------------------------------------------------
 local function main_button_handler(tabview, fields, name, tabdata)
+	local serverlist = {}
```

Comment:
No need to initialize this, if you set it to either `menudata.search_result` or `menudata.favorites` anyway.

Also in this case the entire check could just be abbreviated into:
```lua
local serverlist = menudata.search_result or menudata.favorites
```